plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -556,6 +607,15 @@ class producer_plugin_impl : public std::enable_shared_from_this<producer_plugin
                                          transaction_metadata::trx_type trx_type,
                                          bool return_failure_traces,
                                          next_function<transaction_trace_ptr> next) {
+         if ( trx_type == transaction_metadata::trx_type::read_only && _ro_thread_pool_size > 0 ) {
+            // Parallel read-only trx execution enabled.
+            // Store the transaction in read-only-trx queue so that it is
+            // executed in read window
+            std::lock_guard<std::mutex> g( _ro_trx_queue.mtx );
+            _ro_trx_queue.queue.push_back({trx, std::move(next)});
```

Comment:
I believe this queue size is limited only by `--http-max-in-flight-requests` which defaults to unlimited and also by `--http-max-bytes-in-flight-mb` which defaults to 500MB. That is very roughly something like 3-4 million max read-only trxs that can be queued (if I did the math correctly). It may be worth noting in the PR description read-only queue size is restricted by these two values.

Reply:
A read-only transaction roughly takes about 100 bytes? Will update the PR description.